External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -185,52 +178,81 @@
         end
     else
         % Image contains only NaNs - skip this image
-    end
-
-
+	end
 end
 
-
 %% ========================================================================================
 %% ========================================================================================
 %% METHOD 3
-function [IM] = xASL_im_ExtrapolateLinearlyOverNaNs(IM, nEdgeVoxels)
+function [IM] = xASL_im_ExtrapolateLinearlyOverNaNs(IM, nEdgeVoxels, VoxelSize)
 %xASL_im_ExtrapolateLinearlyOverNaNs
 % The input of this function is 3D (iterations of other dimensions are done
 % in the main function above)
 %
 % IM          - input image
 % nEdgeVoxels - number of voxels at the edges of the non-nan values that we check for weird interpolation values, 
 % to be set to NaN and ignored in the extrapolation
+% VoxelSize   - voxel size is needed for extrapolating correctly as the transformation is given in mm and not voxels
 
 %% 0. Admin
 if nargin<2 || isempty(nEdgeVoxels)
     nEdgeVoxels = 0;
     % We can try this later, but it seems to work less well than doing the same based on the distance transform
 end
 
+if nargin < 3 || isempty(VoxelSize)
```

Comment:
You said you would leave my method intact and only work from 250 onwards ;)

Reply:
Yes. And I did :) I've later told you that cosmetic updates are needed to update your method from 3D to work with the three fields at once :)

I need to keep the voxel here, as this is input parameter handling. So that's why I do a basic check here, and the real estimation later. But I can remove it from here completely if you want.